src/components/page/AddUserPage.tsx

```diff
@@ -0,0 +1,198 @@
+import { useState } from "react";
+import Sidebar from "../common/Sidebar.tsx";
+import Header from "../common/Header.tsx";
+import Footer from "../common/Footer.tsx";
+import apiClient from "../../api/ApiClient.ts";
+import {Link} from "react-router-dom";
+
+const AddUserPage = () => {
+    const [formData, setFormData] = useState({
+        email: "",
+        password: "",
+        firstName: "",
+        lastName: "",
+        role: "",
+    });
+
+    const [error, setError] = useState("");
+    const [success, setSuccess] = useState("");
+    const [emailValid, setEmailValid] = useState(true);
+
+    const handleChange = (e: React.ChangeEvent<HTMLInputElement | HTMLSelectElement>) => {
+        const { id, value } = e.target;
+
+        if (id === "email") {
+            const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+            setEmailValid(emailRegex.test(value));
+        }
+
+        setFormData((prev) => ({
+            ...prev,
+            [id]: value,
+        }));
+    };
+
+    const handleSubmit = async (e: React.FormEvent) => {
+        e.preventDefault();
+
+        if (!emailValid) {
+            setError("Podaj poprawny adres email.");
+            return;
+        }
+
+        try {
+            setError("");
+            setSuccess("");
+
+            await apiClient.post("/admin/users/register", formData);
+
+            setSuccess("Użytkownik został dodany pomyślnie!");
+            setFormData({
+                email: "",
+                password: "",
+                firstName: "",
+                lastName: "",
+                role: "",
+            });
+        } catch (error) {
+            setError(error.response?.data?.message || "Wystąpił błąd podczas dodawania użytkownika.");
+        }
+    };
+
+    const isFormValid =
```

Comment:
This is actualy registerUser dto, I guess it should be unified in terms of validation (for example password min 8 chars)

![2024-12-09_22-45](https://github.com/user-attachments/assets/a888d40d-0a61-431b-8a5a-653ffcff8d9d)